presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1915,6 +1916,11 @@ public SystemSessionProperties(
                         "Rewrite left join with is null check to semi join",
                         featuresConfig.isRewriteExpressionWithConstantVariable(),
                         false),
+                booleanProperty(
+                        PRINT_ESTIMATION_STATS_FROM_CACHE,
+                        "In the end of query optimization, print the estimation stats from cache populated during optimization instead of calculating from ground",
```

Comment:
When printing estimated plan stats after optimization is complete, such as in an EXPLAIN query or for logging in a QueryCompletedEvent, get stats from a cache that was populated during query optimization rather than recalculating the stats on the final plan.